3p-ecosystem/src/main/java/com/google/homesampleapp/screens/inspect/InspectFragment.kt

```diff
@@ -67,15 +78,37 @@ class InspectFragment : Fragment() {
     Timber.d(lifeCycleEvent("onCreateView()"))
 
     // Setup the binding with the fragment.
-    binding = DataBindingUtil.inflate(inflater, R.layout.fragment_inspect, container, false)
+    binding = DataBindingUtil.inflate<FragmentInspectBinding>(
+      inflater,
+      R.layout.fragment_inspect,
+      container,
+      false
+    ).apply {
+      composeView.apply {
+        // Dispose the Composition when the view's LifecycleOwner is destroyed
+        setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)
+        setContent {
+          MaterialTheme {
+            InspectRoute()
+          }
+        }
+      }
+    }
 
     // Setup UI elements and livedata observers.
     setupUiElements()
-    setupObservers()
 
     return binding.root
   }
 
+  override fun onResume() {
+    Timber.d("onResume")
+    super.onResume()
+    // FIXME: right place to do this?
+    // FIXME: first pass we have nothing, then we have something. Possible to make this smoother?
+    viewModel.inspectDevice(selectedDeviceViewModel.selectedDeviceLiveData.value!!.device.deviceId)
```

Comment:
You can do this directly in Compose now using the new lifecycleEffects APIs: https://developer.android.com/topic/libraries/architecture/compose#run-code

Reply:
Thanks for the tip. Used in the latest commit.